packages/react-router-dev/typescript/typegen.ts

```diff
@@ -180,6 +180,10 @@ function parseParams(urlpath: string) {
     .filter((s) => s.startsWith(":"))
     .forEach((param) => {
       param = param.slice(1); // omit leading `:`
+      // If there was an escaped dot, we need to remove it (eg $lang[.]xml => lang.xml => lang)
+      if (param.includes(".")) {
+        param = param.split(".")[0];
+      }
```

Comment:
Haven't tried something like this in a while, but pretty sure the `$` makes the _entire_ segment into a path param, so `/$lang.xml` would match `/blah` no? Which would mean we want to keep `.xml` part in the param too:

```ts
type Params {
  "lang.xml": string
}
```

Reply:
I'm pretty sure it doesn't,  it has to match the path with .xml at the end, eg en.xml matches en doesn't,  I can confirm a bit later if you want me too

Reply:
I think I actually found a bug while trying this, for a route `sitemap.$lang[.]xml.ts` when I try the url `http://localhost:3000/sitemap/blah` the app crashes with the following error in the browser:
![image](https://github.com/user-attachments/assets/309e700f-f515-4742-8031-34b52f0b69b5)
and in the terminal:
![image](https://github.com/user-attachments/assets/ec99c943-8404-4c4a-bbc9-2c786e126060)

The .xml part is required in the wildcard and it only matches routes that end with .xml, so `/blah` is invalid, although it crashes, I think before it didn't use to do that in older versions, I might be wrong though

Reply:
This looks like a bug to me.

That said, for paths that are correctly matched like `en.xml`, what value do you get for the params (e.g. from `useParams`)? `en` or `en.xml`?

Reply:
I get en, .xml is not a part of the wildcard